src/components/popover/Popover.tsx

```diff
@@ -0,0 +1,142 @@
+  const contentRef = useRef<HTMLDivElement | null>(null);
+  const containertRef = useRef<HTMLDivElement | null>(null);
+
+  const isOpen = useMemo(() => (renderOpenEl ? internalOpen : open), [renderOpenEl, internalOpen, open]);
+
+  const anchorPosition = useMemo(() => {
+    const rects = anchorRects ? anchorRects : anchorEl?.current?.getBoundingClientRect();
+
+    return {
+      top: getTopPosition(rects, anchorOriginVertical),
+      left: getLeftPosition(rects, anchorOriginHorisontal)
+    };
+  }, [anchorEl, anchorOriginVertical, anchorOriginHorisontal, anchorRects]);
+
+  const originPosition = useMemo(() => {
+    return {
+      top: getVerticalTranslate(contentRects, transformOriginVertical),
+      left: getHorizontalTranslate(contentRects, transformOriginHorizontal)
+    };
+  }, [transformOriginVertical, transformOriginHorizontal, contentRects]);
+
+  const endPosition = useMemo(() => {
+    return preventOverflow(anchorPosition, originPosition, contentRects, containerRects, safetyMarginUnit);
+  }, [anchorPosition, originPosition, contentRects, containerRects, safetyMarginUnit]);
+
+  const edgeMargins = useMemo(() => {
+    const top = hasTopMargin(anchorOriginVertical, anchorOriginHorisontal, transformOriginVertical);
+    const right = hasRightMargin(anchorOriginVertical, anchorOriginHorisontal, transformOriginVertical);
+    const left = hasLeftMargin(anchorOriginVertical, anchorOriginHorisontal, transformOriginVertical);
+    const bottom = hasRightBottom(anchorOriginVertical, anchorOriginHorisontal, transformOriginVertical);
+
+    return {
+      top: top && edgeMarginUnit,
+      left: right && edgeMarginUnit,
+      right: left && edgeMarginUnit,
+      bottom: bottom && edgeMarginUnit
+    };
+  }, [anchorOriginVertical, anchorOriginHorisontal, transformOriginVertical, edgeMarginUnit]);
+
+  const handleInternalClose = useCallback(() => setInternalOpen(false), []);
+
+  const handleInternalOpen = useCallback(() => setInternalOpen(true), []);
+
+  const hanldeClose = useCallback(() => {
+    renderOpenEl ? handleInternalClose() : onClose();
+    setContentRects(null);
+    setContainerRects(null);
+  }, []);
+
+  const blockClose = useCallback((e: React.SyntheticEvent) => e.stopPropagation(), []);
+
+  const resizeHandler = useCallback(() => (isOpen ? registerResizeHandler() : unRegisterResizeHandler()), [isOpen]);
+
+  const updateDependentRects = useCallback(() => {
+    const anchorRects = renderOpenEl
+      ? renderElRef?.current?.getBoundingClientRect()
+      : anchorEl?.current?.getBoundingClientRect();
+    const containerRects = containertRef?.current?.getBoundingClientRect();
+    const contentRects = contentRef?.current?.getBoundingClientRect();
+
+    anchorRects && setAnchorRects(anchorRects);
+    containerRects && setContainerRects(containerRects);
+    contentRects && setContentRects(contentRects);
+  }, [anchorEl, containertRef]);
+
+  const registerResizeHandler = useCallback(() => {
```

Comment:
All code which add resize handler and remove them move into new hook file called useWindowResize, which accept cb as an argument.